src/operations/aggregate.ts

```diff
@@ -47,14 +42,11 @@ export class AggregateOperation<T = Document> extends CommandOperation<T> {
   pipeline: Document[];
   hasWriteStage: boolean;
 
-  constructor(parent: OperationParent, pipeline: Document[], options?: AggregateOptions) {
-    super(parent, options);
+  constructor(ns: MongoDBNamespace, pipeline: Document[], options?: AggregateOptions) {
+    super(undefined, { ...options, dbName: ns.db });
 
     this.options = options ?? {};
-    this.target =
-      parent.s.namespace && parent.s.namespace.collection
-        ? parent.s.namespace.collection
-        : DB_AGGREGATE_COLLECTION;
+    this.target = ns.collection || DB_AGGREGATE_COLLECTION;
```

Comment:
```suggestion
    this.target = ns.collection ?? DB_AGGREGATE_COLLECTION;
```
just a nit, optional

Reply:
I changed that because when ns.collection is set to the empty string, we set the target to the empty string instead of `DB_AGGREGATE_COLLECTION`, which leads to a few test failures

Reply:
ah Fascinating, I feel like that makes this worthy of more explicit code.

This but with curly braces
```js
if (ns.collection === undefined || ns.collection === '') this.target = DB_AGGREGATE_COLLECTION;
else this.target = ns.collection;
```

Or we can bring back the ternary, Or leave it as is with a comment, `// covers undefined and empty string`, It's just easy to over look if the code itself or a comment doesn't call it out, what do you think?

Reply:
yeah that sounds good. Makes sense to have the comment so the next person to see it and thinks to change it won't have to go bughunting like I did

Reply:
Lol I just noticed this after doing my review... I think a comment would be better; explicit undefined checks are somewhat weird and puts into question why we aren't validating for the other excluded cases... if we're concerned about nullish values but only these ones, why aren't we throwing errors on the other kinds?

Reply:
```
if (typeof ns.collection === 'string' && ns.collection !== '') this.target = ns.collection
else this.target = DB_AGGREGATE_COLLECTION;
```

Ok one more code suggestion ^ 
But also reverting to the ternary and using the same nullish that was there before along with a comment is a good solution to me as well. Your point is correct we should make sure that we are actually filtering for our intention and not leaving a gap that might suggest null is okay here.

Reply:
I'd recommend against the potential behavior change unless truly necessary; and if we do make that behavior change, we'd want to throw on ns.collection not being null/undefined so that we can guard against unexpected input sneaking through (one of those runtime errors)... feels like overkill for this ticket though, hence easier to just keep the original behavior (`this.target = ns.collection || DB_AGGREGATE_COLLECTION;`)